src/aws-lambda/mparticle-personalize/mparticle-personalize.js

```diff
@@ -0,0 +1,177 @@
+                var personalizeCampaignARN = param.Value;
+            }
+        }
+        
+        // Init mParticle libraries for the function invocation
+        var mpApiInstance = new mParticle.EventsApi(new mParticle.Configuration(mpApiKey, mpApiSecret));
+    } catch (e) {
+        console.log("Error getting SSM parameter for loadbalancer.");
+        console.log(e); 
+        throw e;   
+    }
+
+    for (const record of event.Records) {
+        const payloadString = Buffer.from(record.kinesis.data, 'base64').toString('ascii');
+        const payload = JSON.parse(payloadString);
+        const events = payload.events;
+        var amazonPersonalizeUserId;
+        
+        // First, get the mParticle user ID from the events payload.  In this example, mParticle will send all the events
+        // for a particular user in a batch to this lambda.
+        // retreive the mParticle user id which is available for anonymous and known customer profiles
+        var anonymousID = events[0].data.custom_attributes.mpid.toString();
+        
+        // if the customer profile is known then replace the amazon Personalize User id with the actual
+        // personalize Id captured from the user's profile
+        if(payload.user_attributes && payload.user_attributes.amazonPersonalizeId)
+            amazonPersonalizeUserId = payload.user_attributes.amazonPersonalizeId; 
+        else
+            amazonPersonalizeUserId = anonymousID;
+        // Verify in mParticle's payload if there is a customer id set within the customer profile
+        // this will be used for identity resolution later on within mParticle.
+        var customerId = null;
+        if(payload.user_identities){
+                for (const identityRecord of payload.user_identities)
+                {
+                    if(identityRecord.identity_type==="customer_id")
+                        customerId = identityRecord.identity; 
+                }
+            }
+
+        var params = {
+            sessionId: payload.message_id,
+            userId: amazonPersonalizeUserId,
+            trackingId: personalizeTrackerID,
+            eventList: []
+        };
+
+        for (const e of events) {
+            if (e.event_type === "commerce_event" && reportActions.indexOf(e.data.product_action.action) >= 0) {
+                const timestamp = Math.floor(e.data.timestamp_unixtime_ms / 1000);
+                const action = e.data.product_action.action;
+                const event_id = e.data.event_id;
+             
+
+                let params = {
+                    sessionId: payload.message_id,
+                    userId: amazonPersonalizeUserId,
+                    trackingId: personalizeTrackerID,
+                    eventList: []
+                };
+
+                // Build the list of events for the user session...
+                for (const product of e.data.product_action.products) {
+                    const purchasedItem = { itemId: product.id };
```

Comment:
The `discount` column is missing. Should be a `Yes`/`No` value indicating if the product interacted with is discounted or not. Suggest defaulting to `No` if a value cannot be passed through mParticle.

Reply:
added discount Column to set to No